discord/commands/options.py

```diff
@@ -111,6 +116,23 @@ def __init__(self, thread_type: Literal["public", "private", "news"]):
         self._type = type_map[thread_type]
 
 
+AutocompleteReturnType = Union[
+    Iterable["OptionChoice"], Iterable[str], Iterable[int], Iterable[float]
+]
+T = TypeVar("T", bound=AutocompleteReturnType)
+MaybeAwaitable: TypeAlias = Union[T, Awaitable[T]]
+AutocompleteFunction = Union[
+    Callable[[AutocompleteContext], MaybeAwaitable[AutocompleteReturnType]],
+    Callable[
+        [discord.Cog, AutocompleteContext], MaybeAwaitable[AutocompleteReturnType]
+    ],
+    Callable[[AutocompleteContext, Any], MaybeAwaitable[AutocompleteReturnType]],
+    Callable[
+        [discord.Cog, AutocompleteContext, Any], MaybeAwaitable[AutocompleteReturnType]
+    ],
+]
+
+
```

Comment:
This is making the checks fail, so move it to the ``if TYPE_CHECKING`` block (everything else should work)

Reply:
I was already working on it, issue is with discord.Cog.

Reply:
Yeah, doing ``import discord`` on the ``if TYPE_CHECKING`` block should fix it (which would require to move this there)

Reply:
What I did actually was simply `discord.Cog` -> `"discord.Cog"`. But I moved the typing_extensions import to `TYPE_CHECKING` as it is not required by pycord so we should not use it technically.

Reply:
Idk if it's better to move it in `TYPE_CHECKING` entirely, what do you think ?

Reply:
I think it would be better to move it all into the ``TYPE_CHECKING`` block, and also, the ``typing_extensions`` import is not required, as doing ``variable = generic type`` already lets the linter know it is a type alias.

Reply:
we should really be importing Cog directly (`from .cog import Cog`) instead of referencing `discord`